listeners.go

```diff
@@ -180,9 +180,9 @@ func updateListeners(port string, tlsEnabled bool, tlsConfig *tls.Config) error
 		if tlsEnabled && tlsConfig == nil {
 			// "Redirect" server
 			srv = &http.Server{
-				Handler:   http.HandlerFunc(httpsRedirect),
 				TLSConfig: tlsConfig,
 			}
+			srv.Handler = http.HandlerFunc(httpsRedirect(srv.Handler))
```

Comment:
The srv.Handler parameter will just be nil, no? Why not keep the line as it was and specify nil explicitly?

Reply:
Argh, I had expected that to be `http.DefaultHandlerMux`, you're right.